src/Compilers/CSharp/Portable/Errors/DiagnosticInfoWithSymbols.cs

```diff
@@ -20,5 +21,17 @@ internal DiagnosticInfoWithSymbols(bool isWarningAsError, ErrorCode errorCode, o
         {
             this.Symbols = symbols;
         }
+
+        public override bool Equals(object obj)
+        {
+            return obj is DiagnosticInfoWithSymbols diws && 
+                   this.Symbols.Equals(diws.Symbols) && 
+                   base.Equals(obj);
+        }
+
+        public override int GetHashCode()
+        {
+            return Hash.Combine(this.Symbols.GetHashCode(), base.GetHashCode());
```

Comment:
>this.Symbols.GetHashCode() [](start = 32, length = 26)

Here as well, this is probably not based on content of the array, but it should be.